component/component.go

```diff
@@ -190,9 +190,54 @@ func (f CreateDefaultConfigFunc) CreateDefaultConfig() Config {
 	return f()
 }
 
-// InstanceID uniquely identifies a component instance
+// InstanceID uniquely identifies a component instance.
 type InstanceID struct {
-	ID          ID
-	Kind        Kind
-	PipelineIDs map[ID]struct{}
```

Comment:
Potentially these are being used by other components that are not the healthcheck v2 extension (outside of contrib). Since it's quite easy, would you mind keeping these fields but deprecate them for one version?

Reply:
Ah, right :( 

Doing a quick Github search, we do have a bunch of external users of this https://github.com/search?q=%22component.InstanceID%22+language%3Ago+NOT+is%3Afork+NOT+repo%3Aopen-telemetry%2Fopentelemetry-collector+NOT+repo%3Aopen-telemetry%2Fopentelemetry-collector-contrib&type=code

I would be fine with this if we notify grafana/agent and grafana/alloy of this, they seem to be the biggest users